lib/private/Share/Helper.php

```diff
@@ -126,4 +126,29 @@ public static function isSameUserOnSameServer($user1, $server1, $user2, $server2
 
 		return false;
 	}
+
+	public static function getTokenLength(): int {
+		$config = \OC::$server->getConfig();
+		$tokenLengthValue = $config->getAppValue('core', 'shareapi_token_length', (string)self::DEFAULT_TOKEN_LENGTH);
+
+		// Check for 'min', 'max', or non-numeric input
+		if (strtolower($tokenLengthValue) === 'min') {
+			$tokenLength = self::MIN_TOKEN_LENGTH;
+		} elseif (strtolower($tokenLengthValue) === 'max') {
+			$tokenLength = self::MAX_TOKEN_LENGTH;
+		} elseif (!is_numeric($tokenLengthValue) || (int)$tokenLengthValue === 0) {
+			$tokenLength = self::DEFAULT_TOKEN_LENGTH;
+		} else {
+			$tokenLength = (int)$tokenLengthValue;
+		}
```

Comment:
Simplicity is key. Let's simply only allow integers:
```suggestion
		$config = \OCP\Server::get(\OCP\IAppConfig::class);
		$tokenLength = $config->getValueInt('core', 'shareapi_token_length', self::DEFAULT_TOKEN_LENGTH);
		$tokenLength = $tokenLength ?: self::DEFAULT_TOKEN_LENGTH;
```

Reply:
Resolved with https://github.com/nextcloud/server/pull/47265/commits/ee24fdd276df5ec44323aec6105577e623348b01